### PR TITLE
Prepare manifest for new release

### PR DESCRIPTION
**Pull request by Szymongib**
#### Summary

This PR updates the Operator installation manifest before cutting the new release.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-40745

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: docs/mattermost-operator/mattermost-operator.yaml [hunk 1]
```diff
@@ -5650,7 +5696,7 @@ spec:
           value: "20"
         - name: REQUEUE_ON_LIMIT_DELAY
           value: 20s
-        image: mattermost/mattermost-operator:v1.16.0
+        image: mattermost/mattermost-operator:latest
```
Comment (spirosoik) [inline docs/mattermost-operator/mattermost-operator.yaml:5699]: should we point here v1.17?

**Review by spirosoik — APPROVED** [review]
looks good to me except one comment

File: docs/mattermost-operator/mattermost-operator.yaml (hunk 1 above)
Comment (Szymongib) [reply docs/mattermost-operator/mattermost-operator.yaml:5699]: This is autogenerated by kustomize, bump to 1.17 will be done with the next direct commit together with some other version changes.

**Review by gabrieljackson — APPROVED** [review]
👍

**Pull request merged**